source/user/pbc/df.rst

```diff
@@ -3,495 +3,149 @@
-    >>> from pyscf.pbc import gto, df, scf
-    >>> cell = gto.M(atom='He 1 1 1', a=np.eye(3)*2, basis='3-21g')
-    >>> gdf = df.GDF(cell)
-    >>> gdf.kpts = cell.make_kpts([2,2,2])
-    >>> gdf.get_eri([kpts[0],kpts[0],kpts[1],kpts[1]])
-
-In the mean-field calculation, assigning :attr:`kpts` attribute to mean-field
-object updates the :attr:`kpts` attribute of the underlying DF method::
-
-    >>> import numpy as np
-    >>> from pyscf.pbc import gto, df, scf
-    >>> cell = gto.M(atom='He 1 1 1', a=np.eye(3)*2, basis='3-21g')
-    >>> mf = scf.KRHF(cell).density_fit()
-    >>> kpts = cell.make_kpts([2,2,2])
-    >>> mf.kpts = kpts
-    >>> mf.with_df.get_eri([kpts[0],kpts[0],kpts[1],kpts[1]])
-
-Once the GDF integral tensor was initialized, the :class:`GDF` can be only used
-with certain k-points calculations.  An incorrect :attr:`kpts` argument can lead
-to a runtime error::
-
-    >>> import numpy as np
-    >>> from pyscf.pbc import gto, df, scf
-    >>> cell = gto.M(atom='He 1 1 1', a=np.eye(3)*2, basis='3-21g')
-    >>> gdf = df.GDF(cell, kpts=cell.make_kpts([2,2,2]))
-    >>> kpt = np.random.random(3)
-    >>> gdf.get_eri([kpt,kpt,kpt,kpt])
-    RuntimeError: j3c for kpts [[ 0.53135523  0.06389596  0.19441766]
-     [ 0.53135523  0.06389596  0.19441766]] is not initialized.
-    You need to update the attribute .kpts then call .build() to initialize j3c.
-
-The GDF initialization is very expensive.  To reduce the initialization cost in
-a series of calculations, it would be useful to cache the GDF integral tensor in
-a file then load them into the calculation when needed.  The GDF integral tensor
-can be saved and loaded the same way as we did for the molecular DF method (see
-:ref:`sl_cderi`)::
+*Modules*: :mod:`df`, :mod:`pbc.df`
 
-    import numpy as np
-    from pyscf.pbc import gto, df, scf
-    cell = gto.M(atom='He 1 1 1', a=np.eye(3)*2, basis='3-21g')
-    gdf = df.GDF(cell, kpts=cell.make_kpts([2,2,2]))
-    gdf._cderi_to_save = 'df_ints.h5'  # To save the GDF integrals
-    gdf.build()
+Introduction
+============
 
-    mf = scf.KRHF(cell, kpts=cell.make_kpts([2,2,2])).density_fit()
-    mf.with_df._cderi = 'df_ints.h5'   # To load the GDF integrals
-    mf.run()
+Like for molecular calculations, density fitting (DF) is useful to reduce the computational cost of manipulating the electron repulsion integrals (ERIs) in periodic calculations. Currently, PySCF provides three periodic DF methods that are different in the auxiliary basis functions being used.
 
+* The fast Fourier transform density fitting (:class:`FFTDF`) uses plane waves (PWs) as the auxiliary basis functions. FFTDF is also known as the Gaussian and plane wave (GPW) approach in literature and other software packages. :cite:`VandeVondele05CPC,Kuhne20JCP`
 
-Auxiliary Gaussian basis
-------------------------
+* The Gaussian density fitting (:class:`GDF`) uses Gaussian-type orbitals (GTOs) as the auxiliary basis functions. :cite:`Sun17JCP`
```

Comment:
I think this should be the first item

Reply:
I put FFTDF first bc it's the *default* used by the periodic SCF module (i.e., the default is not "without DF" but FFTDF). What do you think?